src/main/java/seedu/address/model/startup/Valuation.java

```diff
@@ -0,0 +1,118 @@
+package seedu.address.model.startup;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Startup's valuation in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidValuation(String)}
+ */
+public class Valuation {
+
+    public static final String MESSAGE_CONSTRAINTS =
+        "Valuation of a company must be a INTEGER "
+        + "(No decimal places, i.e. 1 is a integer, 1.1 is not)"
+        + " above -1 and "
+        + " below 5 trillion (the highest valued company in the world is only worth ~3T).";
+    public static final String VALIDATION_REGEX = "\\d+";
+    private static char[] endOfNumber = new char[]{'k', 'm', 'b', 't'};
+
+    public final String value;
+
+    /**
+     * Constructs a {@code Valuation}.
+     *
+     * @param valuation A valid valuation.
+     */
+    public Valuation(String valuation) {
+        requireNonNull(valuation);
+        checkArgument(isValidValuation(valuation), MESSAGE_CONSTRAINTS);
+        value = valuation;
+    }
+
+    /**
+     * Reformats valuation for clean display on the UI, truncates to 4 characters.
+     * @param valuation The valuation to reformat
+     * @return The truncated, short-form version of the valuation.
+     */
+    public static String reformatValuation(String valuation) {
+        Double doubleValuation = Double.valueOf(valuation);
+        // If it's below 1000, we don't need to format it!
+        if (doubleValuation.compareTo((double) 1000) < 0) {
+            return valuation;
+        }
+        return reformatString(doubleValuation, 0);
+    }
+
+    /**
+     * This function helps us reformat the user's input into
+     * a short string representation (within 4 characters).
+     * Credit to Ilya Saunkin, SWE from AWS, shared in the following stackoverflow discussion:
+     * https://stackoverflow.com/questions/4753251/how-to-go-about-formatting-1200-to-1-2k-in-java
+     *
+     * Recursive implementation,
+     * invokes itself for each factor of a thousand,
+     * increasing the class on each invocation.
+     *
+     * @param n the number to format
+     * @param iteration in fact this is the class from the array c
+     * @return a String representing the number n formatted in a cool looking way.
+     */
+    private static String reformatString(double n, int iteration) {
```

Comment:
Not sure how accurate valuation needs to be. Just take note that the function will round off 999999999 to 999m, instead of 1b

Reply:
Yup. I think a round down should be ok, because at such a magnitude 999m vs 1b might not be too big of a gap, especially since the user, or for the purpose of the contact book, might be thinking of looking at companies at a range of valuation instead.